File: BookStore-Angular-MongoDb/angular/src/app/author/models/create-author-dto.ts

export class CreateAuthorDto  {
  name: string;
  birthDate: string;
  shortBio: string;

  constructor(initialValues: Partial<CreateAuthorDto> = {}) {
    if (initialValues) {
      for (const key in initialValues) {
        if (initialValues.hasOwnProperty(key)) {
          this[key] = initialValues[key];
        }
      }
    }
  }
}

File: BookStore-Blazor-EfCore/src/Acme.BookStore.Application.Contracts/Authors/CreateAuthorDto.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Acme.BookStore.Authors;

[Serializable]
public class CreateAuthorDto
{
    [Required]
    [StringLength(AuthorConsts.MaxNameLength)]
    public string Name { get; set; }

    [Required]
    public DateTime BirthDate { get; set; }

    public string ShortBio { get; set; }
}
